wp-dependency-installer.php

```diff
@@ -208,10 +211,34 @@ public function apply_config() {
 				 * @param string $download_link Download link.
 				 * @param array  $dependency    Dependency configuration.
 				 */
-				$download_link = apply_filters( 'wp_dependency_download_link', $download_link, $dependency );
+				$this->config[ $slug ]['download_link'] = apply_filters( 'wp_dependency_download_link', $download_link, $dependency );
 
-				$this->config[ $slug ]['download_link'] = $download_link;
+				if ( ! isset( $dependency['activate_notice'] ) ) {
+					$this->config[ $slug ]['activate_notice'] = true;
+				}
+				if ( ! isset( $dependency['install_notice'] ) ) {
+					$this->config[ $slug ]['install_notice'] = true;
+				}
+
+				/**
+				 * Allow filtering of individual dependency config.
+				 *
+				 * @since 3.0.0
+				 *
+				 * @param array  $dependency    Dependency configuration.
+				 */
+				$this->config[ $slug ] = apply_filters( "wp_dependency_config_{$slug}", $this->config[ $slug ] );
 			}
+
+			/**
+			 * Allow filtering of whole dependencies config.
+			 *
+			 * @since 3.0.0
+			 *
+			 * @param array  $this->config   Dependencies configuration.
+			 */
+			$this->config = apply_filters( 'wp_dependency_config', $this->config );
+
```

Comment:
And what do the above filters have to do with admin notices?

Reply:
For example, they allows you to alter this default values:

https://github.com/afragen/wp-dependency-installer/pull/60/files#diff-5bfef90facabe22b651bbd28d639df25R220

https://github.com/afragen/wp-dependency-installer/pull/60/files#diff-5bfef90facabe22b651bbd28d639df25R217

**NB** current code already works like this (ie. default true):

```php
} elseif ( $this->is_installed( $slug ) ) {
  if ( $is_required ) {
    $this->notices[] = $this->activate( $slug );
  } else {  // <-- same as default: true
    $this->notices[] = $this->activate_notice( $slug );
  }
} else {
  if ( $is_required ) {
    $this->notices[] = $this->install( $slug );
  } else { // <-- same as default: true
    $this->notices[] = $this->install_notice( $slug );
  }
}
```